tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc

```diff
@@ -1506,36 +1506,139 @@ Status Converter::RenameAndMarkOutputTensors(
+    // TODO: accepting a list from the environment variable
+    if (trt_algorithm_id != null_tactic) {
+      allowed_tactics.insert(trt_algorithm_id);
+    }
+
+#if !IS_TRT_VERSION_GE(8, 0, 0, 0)
+    disallowed_tactics = {
+      // turing_fp16_s1688cudnn_fp16_128x128_ldg8_relu_f2f_exp_medium_nhwc_gelu_tn_v1
+      -5927686925093575778,
+      //turing_fp16_s1688cudnn_fp16_128x128_ldg8_relu_f2f_exp_interior_nhwc_gelu_tn_v1
+      -3848538574386518527,
+      //turing_fp16_s1688cudnn_fp16_128x128_ldg8_relu_f2f_exp_small_nhwc_gelu_tn_v1
+      -959009792490796596
+    };
+#endif  // !IS_TRT_VERSION_GE(8, 0, 0, 0)
+
+    // Check whether allowed and disallowed tactics are disjointed.
+    if (allowed_tactics.size() > 0 && disallowed_tactics.size() > 0) {
+      std::vector<int64_t> intersection;
+      for (int64_t allowed : allowed_tactics) {
+        if (disallowed_tactics.count(allowed) > 0) {
+          intersection.emplace_back(allowed);
+        }
+      }
+      if (intersection.size() > 0) {
+        VLOG(1) << "Intersection of allowed and disallowed tactics found "
+                   "non-empty.";
+
+        VLOG(1) << "List of allowed tactics:";
+        VLOG(1) << absl::StrJoin(allowed_tactics, ", ");
+
+        VLOG(1) << "List of disallowed tactics:";
+        VLOG(1) << absl::StrJoin(disallowed_tactics, ", ");
 
+        VLOG(1) << "List of tactics common to both allowed and disallowed "
+                   "tactics:";
+        VLOG(1) << absl::StrJoin(intersection, ", ");
+
+        VLOG(1) << "Aborting.";
+        assert(0);
+      }
+    }
+  }
   // Returns value in [0, nbChoices] for a valid algorithm.
   int32_t selectAlgorithms(const nvinfer1::IAlgorithmContext& algoContext,
                            const nvinfer1::IAlgorithm* const* algoChoices,
                            int32_t nbChoices, int32_t* selection) override {
+
     // TensorRT always provides more than zero number of algorithms
     // in selectAlgorithms.
     assert(nbChoices > 0);
 
-    // making sure that the requested TRT algorithm ID doesn't go above the
-    // max value accepted.
-    selection[0] = std::min(algorithm_id_, nbChoices);
-    return 1;
+    if (allowed_tactics.empty()) {
+      // Disable all disallowed tactics
+      int32_t nbSelections = 0;
+      for (auto i = 0; i < nbChoices; i++) {
+        bool reject{false};
+
+        if (algoChoices[i] == nullptr) {
+          continue;
+        }
+
+        int64_t tacticId = algoChoices[i]->getAlgorithmVariant().getTactic();
+        int64_t implementation =
+            algoChoices[i]->getAlgorithmVariant().getImplementation();
+        nvinfer1::TensorFormat format =
+            algoChoices[i]->getAlgorithmIOInfo(0).getTensorFormat();
+        assert(tacticId != static_cast<int64_t>(Tactic::kINVALID_TACTIC));
+
+        if (disallowed_tactics.find(tacticId) != disallowed_tactics.end()) {
+          reject = true;
+        }
+
```

Comment:
We can more part of the code below here:

if (reject) {
    VLOG(1) << "Rejecting a disallowed tactic: " << tacticId
                  << " for node " << algoContext.getName();
   continue;
}